plugins/scaffolder/src/api.ts

```diff
@@ -14,14 +14,53 @@
  * limitations under the License.
  */
 
-import { createApiRef, DiscoveryApi, IdentityApi } from '@backstage/core';
+import {
+  createApiRef,
+  DiscoveryApi,
+  Observable,
+  IdentityApi,
+} from '@backstage/core';
+import ObservableImpl from 'zen-observable';
+import { ScaffolderTask, Status } from './types';
 
 export const scaffolderApiRef = createApiRef<ScaffolderApi>({
   id: 'plugin.scaffolder.service',
   description: 'Used to make requests towards the scaffolder backend',
 });
 
-export class ScaffolderApi {
+export type LogEvent = {
+  type: 'log' | 'completion';
+  body: {
+    message: string;
+    stepId?: string;
+    status?: Status;
+  };
+  createdAt: string;
+  id: string;
+  taskId: string;
+};
+
+export interface ScaffolderApi {
+  /**
+   * Executes the scaffolding of a component, given a template and its
+   * parameter values.
+   *
+   * @param templateName Template name for the scaffolder to use. New project is going to be created out of this template.
```

Comment:
```suggestion
   * @param templateName Name of the Template entity for the scaffolder to use. New project is going to be created out of this template.
```

Reply:
But at this point ... we should pass in an `EntityName` (from catalog-model) while we're at it so we actually can extend in the future elsewhere to have namespaces. If we don't have backend support for it yet, just throw an error if the caller tries to pass in anything except kind Template and namespace default